### PR TITLE
 Instructor student search: '(' causes server error #4596 

### LINKED ISSUE #4596: instructor student search: '(' causes server error
searching for "adam (jin)" causes a server error. I was searching for a student name who also had an alias, given in brackets.

### PR DESCRIPTION
**Pull request by rclakmal**
Fixes #4596

### CODE REVIEW

**Comment by rclakmal** [comment]
Some background on the fix, Issue occurs not only for brackets but also for other special characters. Since we cannot make sure that when these special characters be included in the search query, we need to enclose each key word with quotes.
https://cloud.google.com/appengine/docs/java/search/query_strings

**Comment by damithc** [comment]
@taniach is there any way to cover this case in tests?

**Comment by taniach** [comment]
We can add a new test to AdminSearchPageUiTest.

@rclakmal Can you add it in this pull request?

**Comment by rclakmal** [comment]
@taniach Will do

**Comment by rclakmal** [comment]
Test case added.

File: src/test/java/teammates/test/cases/ui/browsertests/AdminSearchPageUiTest.java [hunk 1]
```diff
@@ -73,6 +73,16 @@ private void testSearch() {
         
         assertTrue(isSearchPanelPresent());
         assertTrue(isSearchDataDisplayCorrect());
+        
+        ______TS("search for studentNameWithSpecialCharacters");
```
Comment (damithc) [inline src/test/java/teammates/test/cases/ui/browsertests/AdminSearchPageUiTest.java]: `studentNameWithSpecialCharacters` need not be camelCase.

**Comment by damithc** [comment]
Minor cosmetic change requested. Comment when done.

**Comment by rclakmal** [comment]
Modified the test description

**Comment by thyageshm** [comment]
@taniach, I believe you have to help to merge this one :-)

**Pull request merged**